beets/util/__init__.py

```diff
@@ -686,7 +716,7 @@ def _legalize_stage(path, replacements, length, extension, fragment):
 
     # Encode for the filesystem.
     if not fragment:
-        path = bytestring_path(path)
+        path = bytestring_path(path)  # type: ignore
```

Comment:
Hopefully once I get the pathlib PR going this won't be an issue. This was mostly for mypy since it doesn't like when you change the type of named variables, it always throws an error.

Reply:
mypy complains for good reason, changing the type in this way really seems like a recipe for disaster. I'm not sure that pathlib magically makes this disappear, I suspect that the callers of this function might still need a different type depending on whether dealing with path fragments or an entire path.